card/mod/standard/set/all/links.rb

```diff
@@ -80,79 +71,126 @@ def view_link text, view, opts={}
   end
 
   def related_link name_or_card, opts={}
-    name =
-      case name_or_card
-      when Symbol then Card.fetch(name_or_card, skip_modules: true).cardname
-      when Card   then name_or_card.cardname
-      else             name_or_card
-      end
+    name = Card::Name.cardish name_or_card
     opts[:path_opts] ||= { view: :related }
     opts[:path_opts][:related] = { name: "+#{name}" }
     opts[:path_opts][:related].merge! opts[:related_opts] if opts[:related_opts]
     view_link(opts[:text] || name, :related, opts)
   end
 
-  def path opts={}
-    if opts[:action] == :new && opts[:type] &&
-       !(opts[:name] || opts[:card] || opts[:id])
-      opts.delete(:action)
-      base = "new/#{opts.delete(:type)}"
-    else
-      name = opts.delete(:name) || card.name
-      base = opts[:action] ? "card/#{opts.delete :action}/" : ""
+  # @param opts [Hash]
+  # @option opts [Symbol] :action card action (:create, :update, :delete)
+  # @option opts [Integer, String] :id
+  # @option opts [String, Card::Name] :name
+  # @option opts [String] :type
```

Comment:
What about this: We get rid of type and use `action: :new, mark: `cardtype` and if cardtype is not a cardtype we produce `/new/Basic?card[name]=cardtype`